libtransmission/rpc-server.cc

```diff
@@ -587,38 +705,48 @@ static void startServer(void* vserver)
         return;
     }
 
-    struct evhttp* httpd = evhttp_new(server->session->event_base);
+    struct event_base* base = server->session->event_base;
+    struct evhttp* httpd = evhttp_new(base);
+
     evhttp_set_allowed_methods(httpd, EVHTTP_REQ_GET | EVHTTP_REQ_POST | EVHTTP_REQ_OPTIONS);
 
     char const* address = tr_rpcGetBindAddress(server);
 
     tr_port const port = server->port;
 
-    if (evhttp_bind_socket(httpd, address, port) == -1)
+    bool success = false;
+    if (server->bindAddress.type == TR_RPC_AF_UNIX)
+    {
+        success = bindUnixSocket(base, httpd, address, server->rpc_socket_mode);
+    }
+    else
+    {
+        success = evhttp_bind_socket(httpd, address, port) != -1;
+    }
```

Comment:
`success` can be made const:

```suggestion
    bool const success = server->bindAddress.type == TR_RPC_AF_UNIX)
        ? bindUnixSocket(base, httpd, address, server->rpc_socket_mode)
        : (evhttp_bind_socket(httpd, address, port) != -1);
```